pkg/controller/namespace/namespace_controller.go

```diff
@@ -451,6 +461,21 @@ func (r *ReconcileNamespace) deleteRuntime(namespace *corev1.Namespace) error {
 
 	return nil
 }
+func (r *ReconcileNamespace) unbindWorkspace(instance *corev1.Namespace) error {
+	changed := false
+	for i := 0; i < len(instance.OwnerReferences); i++ {
```

Comment:
this is not the straightforward way, not easy to read, I suggest make a copy without unwanted references, and then assign back to instance